common_python_scripts/partition_transcripts.py

```diff
@@ -0,0 +1,75 @@
+import pandas as pd
+import geopandas as gpd
+from shapely.geometry import Point
+import numpy as np
+import argparse
+
+def main(transcript_file, cell_polygon_file):
+
+    def find_containing_polygon(transcript):
+        point = transcript.geometry
+        possible_matches_index = list(cell_polygons_sindex.query(point, predicate='intersects'))
+        
+        if len(possible_matches_index) == 0:
+            return None
+        
+        return possible_matches_index[0]
+
+    def process_chunk(transcript_chunk, cell_polygons_gdf, cell_polygons_sindex):
+        if not transcript_chunk.empty:
+            transcript_chunk['cell_index'] = transcript_chunk.apply(find_containing_polygon, axis=1)
+            transcript_chunk['color'] = transcript_chunk['cell_index'].apply(lambda x: cell_polygons_gdf.loc[x, 'color'] if pd.notna(x) else 'grey')
+        else:
+            transcript_chunk['cell_index'] = None
+            transcript_chunk['color'] = 'grey'
+        
+        return transcript_chunk
+
+    transcript_df = pd.read_csv(transcript_file)
+    transcript_df['geometry'] = transcript_df.apply(lambda row: Point(row['global_x'], row['global_y']), axis=1)
+    transcripts_gdf = gpd.GeoDataFrame(transcript_df, geometry='geometry')
+
+    cell_polygons_gdf = gpd.read_parquet(cell_polygon_file)
+
+    cell_polygons_gdf.reset_index(inplace=True)
+    cell_polygons_gdf.drop(['index', 'shard', 'job', '0'], axis=1, inplace=True)
+    cell_polygons_gdf.rename(columns={'level_0': 'cell_ids'}, inplace=True)
+
+    cell_polygons_sindex = cell_polygons_gdf.sindex
+
+    color_list = ['red', 'blue', 'green', 'yellow', 'purple', 'black', 'orange', 'grey', 'pink', 'brown']
```

Comment:
we can get rid of color_list since we are not going to be using this in the partitioning step.